File: src/neetcode/Java/_11ContainerWithMostWater.java
package neetcode.Java;

public class _11ContainerWithMostWater {
    public static int maxArea(int[] height) {
        int s=0,e=height.length-1;
        int maxLevel=0,lev=0;
        while (s<e){
            lev=(e-s)*Math.min(height[s],height[e]);
            if(lev>maxLevel) maxLevel=lev;
            if(height[e]>height[s]) s++;else e--;
        }
        return maxLevel;
    }

    public static void main(String[] args) {
        int[] arr={1,1};
        int res = maxArea(arr);
        System.out.println(res);
    }
}

File: src/neetcode/kotlin/_11ContainerWithMostWater.kt
package neetcode.kotlin

import kotlin.math.min

object _11ContainerWithMostWater {
    fun maxArea(height: IntArray): Int {
        var s = 0
        var e = height.size - 1
        var maxLevel = 0
        var lev = 0
        while (s < e) {
            lev = ((e - s) * min(height[s].toDouble(), height[e].toDouble())).toInt()
            if (lev > maxLevel) maxLevel = lev
            if (height[e] > height[s]) s++ else e--
        }
        return maxLevel
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val arr = intArrayOf(1, 1)
        val res = maxArea(arr)
        println(res)
    }
}
